src/harness/testcases/WINNF_FT_S_REG_testcase.py

```diff
@@ -424,6 +424,165 @@ def test_WINNF_FT_S_REG_5(self):
+    device_2['cbsdSerialNumber'] = 's' * 65
+    # invalid_fcc_id with length > 20 octets
+    device_3['fccId'] = 'f' * 21
+    device_4['userId'] = 'invalid_userId_@'
+    device_5['installationParam']['latitude'] = 90.01
+    device_6['measCapability'] = ['invalid_measCapability']
+    device_7['installationParam']['eirpCapability'] = 48
+    # Convert device_7's registration request to embed cpiSignatureData
+    convertRequestToRequestWithCpiSignature(cpi_private_key, cpi_id,
+                                            cpi_name, device_7)
+    device_8['installationParam']['latitude'] = 38.882162
+    device_8['installationParam']['longitude'] = -77.113755
+    device_8['installationParam']['height'] = 4.0
+    device_8['installationParam']['heightType'] = 'AGL'
+    device_8['installationParam']['indoorDeployment'] = False  
+    device_9['installationParam']['eirpCapability'] = 31
+    device_10['installationParam']['indoorDeployment'] = True
+    device_10['installationParam']['eirpCapability'] = 31
+    
+    # Convert device_10's registration request to embed cpiSignatureData
+    convertRequestToRequestWithCpiSignature(cpi_private_key, cpi_id,
+                                            cpi_name, device_10)
+    installation_param_device_11 = device_11['installationParam']
+    # Convert device_11's registration request to embed cpiSignatureData
+    convertRequestToRequestWithCpiSignature(cpi_private_key, cpi_id,
+                                            cpi_name, device_11)
+    # Re-add installationParam to the device 11
+    device_11['installationParam'] = installation_param_device_11
+    # Convert device_12's' registration request to embed cpiSignatureData
+    convertRequestToRequestWithCpiSignature(cpi_private_key,\
+                                'Incorrent_installer_id_1', cpi_name, device_12)
+    device_13['installationParam']['latitude'] = 38.882162
+    device_13['installationParam']['longitude'] = -77.013055
+    device_13['installationParam']['height'] = 5.0
+    # Convert device_13's' registration request to embed cpiSignatureData    
+    convertRequestToRequestWithCpiSignature(cpi_private_key,\
+                                cpi_id, cpi_name, device_13)
+    
+    device_15['fccId'] =  'fccId_approved_eirp'
+    device_15['installationParam']['eirpCapability'] = 26
+    
+    device_16['fccId'] =  'fccId_approved_eirp'
+    device_16['installationParam']['eirpCapability'] = 26
+    convertRequestToRequestWithCpiSignature(cpi_private_key,\
+                                cpi_id, cpi_name, device_16)  
+    # Register devices
+    devices = [device_1, device_2, device_3, device_4, device_5, device_6,\
+               device_7, device_8, device_9, device_10, device_11, device_12,\
+               device_13, device_14, device_15, device_16 ]
+    request = {'registrationRequest': devices}
+    response = self._sas.Registration(request)
+    # Check registration response
+   
+    self.assertEqual(
```

Comment:
We also need to assert that a valid cbsdId is included for the first CBSD as per the test spec.

I think it would also be better (though not explicitly stated by the test spec ) to ensure that a cbsdId is included for thirteenth and not included for the rest of the failed registrations.

Reply:
thank you for the comment, the test spec doesn't mention doing any verification regarding the cbsd Id  for this test, and I think the SAS will fail in other REG tests if it behaves differently.
I find your proposed checks logical but actually I removed them after I had added them,
because I think it is better to be aligned with the test spec otherwise we need to add a comment in the code to tag it as unapproved part, right?

Reply:
we agree with your reasoning and we don’t need to add a check for the thirteenth CBSD. However the spec mentions that the response for the first CBSD has to be checked if it has a valid cbsdId.

Reply:
Please hold on the merge, we just noticed that the following comment has not yet be addressed yet. 

We need either do
1.	Suggest to change the spec. to remove SAS response includes a valid cbsdId for the first CBSD. 
Or do
2.	Add the check for validating the cbsdId in the response for the first CBSD

Reply:
thanks, yes I agree with your proposal, I added verification for "valid" Cbsd Id of device 1.
could you check it?